src/python/CRABClient/Commands/report.py

```diff
@@ -351,37 +348,26 @@ def getLumisToProcess(self, userWebDirURL, numJobs, workflow):
         """
         res = {}
         if userWebDirURL:
-            curl = self.prepareCurl()
-            fp = tempfile.NamedTemporaryFile()
-            curl.setopt(pycurl.WRITEFUNCTION, fp.write)
-            hbuf = StringIO.StringIO()
-            curl.setopt(pycurl.HEADERFUNCTION, hbuf.write)
+            url = userWebDirURL + "/run_and_lumis.tar.gz"
+            tarFilename = os.path.join(self.requestarea, 'results/run_and_lumis.tar.gz')
             try:
-                url = userWebDirURL + "/run_and_lumis.tar.gz"
-                curl.setopt(pycurl.URL, url.encode("ascii", "ignore"))
-                self.myPerform(curl, url)
-                header = ResponseHeader(hbuf.getvalue())
-                if header.status == 200:
-                    fp.seek(0)
-                    tarball = tarfile.open(fp.name)
-                    try:
-                        for jobid in xrange(1, numJobs+1):
-                            filename = "job_lumis_%d.json" % (jobid)
+                getFileFromURL(url, filename=tarFilename, proxyfilename=self.proxyfilename)
+                with tarfile.open(tarFilename) as tarball:
+                    for jobid in xrange(1, numJobs+1):
+                        filename = "job_lumis_%d.json" % (jobid)
+                        try:
+                            member = tarball.getmember(filename)
+                        except KeyError:
+                            self.logger.warning("File %s not found in run_and_lumis.tar.gz for task %s" % (filename, workflow))
+                        else:
+                            fd = tarball.extractfile(member)
                             try:
-                                member = tarball.getmember(filename)
-                            except KeyError:
-                                self.logger.warning("File %s not found in run_and_lumis.tar.gz for task %s" % (filename, workflow))
-                            else:
-                                fd = tarball.extractfile(member)
-                                try:
-                                    res[str(jobid)] = json.load(fd)
-                                finally:
-                                    fd.close()
-                    finally:
-                        tarball.close()
-            finally:
-                fp.close()
-                hbuf.close()
+                                res[str(jobid)] = json.load(fd)
+                            finally:
+                                fd.close()
+            except ClientException as ce:
```

Comment:
One small detail, if you pick `LOGGERS['CRAB3']` from https://github.com/dmwm/CRABClient/blob/master/src/python/CRABClient/ClientUtilities.py#L96 it will be possible to print the exception traceback only in the logfile. It might be useful imho.